CardScan/Classes/API.swift

```diff
@@ -135,6 +135,62 @@ public struct Api {
         apiCall(endpoint: endpoint, parameters: apiParameters, completion: completion)
     }
     
+    static public func getModelDownloadConfig(endpoint: String, parameters: [String: Any], completion: @escaping ApiCompletion) {
+        if baseUrl == nil || apiKey == nil {
+            DispatchQueue.main.async { completion(nil, apiUrlNotSet) }
+            return
+        }
+        
+        modelDownloadConfig(endpoint: endpoint, parameters: parameters, completion: completion)
+    }
+    
+    private static func modelDownloadConfig(endpoint: String, parameters: [String: Any], completion: @escaping ApiCompletion) {
+        guard let baseUrl = self.baseUrl else {
+            DispatchQueue.main.async { completion(nil, apiUrlNotSet) }
+            return
+        }
+        
+        guard let url = urlWithQueryParameters(baseUrl: baseUrl, endpoint: endpoint, parameters: parameters) else {
+            DispatchQueue.main.async { completion(nil, defaultError) }
+            return
+        }
+
+        let session = URLSession(configuration: configuration())
+        session.dataTask(with: url) { data, response, error in
+            guard let rawData = data else {
+                DispatchQueue.main.async { completion(nil, defaultError) }
+                return
+            }
+
+            let jsonData = try? JSONSerialization.jsonObject(with: rawData)
+            guard let responseData = jsonData as? [String: Any] else {
+                DispatchQueue.main.async { completion(nil, defaultError) }
+                return
+            }
+            
+            DispatchQueue.main.async {
+                if "error" == responseData["status"] as? String {
+                    completion(nil, ApiError(response: responseData))
+                } else {
+                    completion(responseData, nil)
+                }
+            }
+        }.resume()
+    }
+    
+    static func urlWithQueryParameters(baseUrl: String, endpoint: String, parameters: [String: Any]) -> URL? {
```

Comment:
I'd add a comment that says something like "we are going to cast all of the parameter values to strings so make sure that you don't have any arrays, dicts, optionals, or other values that won't convert cleanly to strings"